src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/multi/SalesforceMultiInputFormat.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce.plugin.source.batch.multi;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.sforce.async.BulkConnection;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.salesforce.SObjectDescriptor;
+import io.cdap.plugin.salesforce.plugin.source.batch.base.SalesforceBaseInputFormat;
+import io.cdap.plugin.salesforce.plugin.source.batch.util.SalesforceSourceConstants;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Input format class which generates input splits for each given query
+ * and initializes appropriate record reader.
+ */
+public class SalesforceMultiInputFormat extends SalesforceBaseInputFormat {
+
+  private static final Gson GSON = new Gson();
+  private static final Type QUERIES_TYPE = new TypeToken<List<String>>() { }.getType();
+
+  @Override
+  public List<InputSplit> getSplits(JobContext context) {
+    Configuration configuration = context.getConfiguration();
+    List<String> queries = GSON.fromJson(configuration.get(SalesforceSourceConstants.CONFIG_QUERIES), QUERIES_TYPE);
```

Comment:
Instead of having separate Multi classes for the input format, record reader, split, and inputformat provider, it is much simpler to just add the query to the SalesforceSplit and enhance the existing inputformat to support multiple queries. The multi source would just configure it with multiple queries while the single source would configure it with a single query.

Reply:
Done, though I felt code was less coupled with previous design.